### PR TITLE
Setup APIGW CloudWatch role via custom resource

### PR DESCRIPTION
**Pull request by medikoo**
APIGW CloudWatch role needs to be assigned. so logs for APIGW can succesfully be wrirtten.

Still APIGW Cloudwatch role is global per account & region combination, which makes it error-prone if handled via CloudFormation stacks. Situation is as follows:
- For every stack new IAM role is created (not breaking unless we're within AWS limits)
- On creation of every stack, eventual previous APIGW CloudWatch role assignement is overriden with new one (not breaking)
- On removal of stack, the role is removed, which makes CloudWatch role assignment broken for remaining stacks (breaking).

Proposed solution derives Cloudwatch role assignment to custom resource, which:
- Creates and assigns new role, if there's no CloudWatch role assigned yet for given region
- Ensures that dedicated role, is accompanied with policy that allows to write CloudWatch logs

Tested by deploying new services, and by deploying services deployed previously with older version of SLS.

Updated integration tests, so it confirms that logs are written

**_Is it a breaking change?:_** NO

### CODE REVIEW

File: lib/plugins/aws/customResources/index.js [hunk 1]
```diff
@@ -47,9 +47,18 @@ function addCustomResourceToService(resourceName, iamRoleStatements) {
...
+  } else if (resourceName === 'apiGatewayCloudWatchRole') {
+    FunctionName = `${funcPrefix}-${this.provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleHandlerFunctionName()}`;
+    Handler = 'apiGatewayCloudWatchRole/handler.handler';
+    customResourceFunctionLogicalId = this.provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleHandlerFunctionLogicalId();
   } else {
     return BbPromise.reject(`No implementation found for Custom Resource "${resourceName}"`);
   }
+  if (FunctionName.length > 64) {
+    return BbPromise.reject(
+      new Error(`Resolved custom resource function name '${FunctionName}' is too long`)
```
Comment (alexdebrie) [inline lib/plugins/aws/customResources/index.js]: Would this happen if the service + stage name combo is too long? Is there anything we could do to help the user, or are they stuck?

File: lib/plugins/aws/customResources/resources/apiGatewayCloudWatchRole/handler.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+function update() {
+  // No actions
+}
+
+function remove() {
+  // No actions
+}
+
+module.exports = {
+  handler: handlerWrapper(handler, 'CustomResouceApiGatewayAccountCloudWatchRole'),
```
Comment (alexdebrie) [inline lib/plugins/aws/customResources/resources/apiGatewayCloudWatchRole/handler.js]:
```suggestion
  handler: handlerWrapper(handler, 'CustomResourceApiGatewayAccountCloudWatchRole'),
```

File: lib/plugins/aws/lib/naming.js [hunk 3]
```diff
@@ -504,4 +498,18 @@ module.exports = {
   getCustomResourceEventBridgeResourceLogicalId(functionName, idx) {
     return `${this.getNormalizedFunctionName(functionName)}CustomEventBridge${idx}`;
   },
+  // API Gateway Account Logs Write Role
+  getCustomResourceApiGatewayAccountCloudWatchRoleHandlerFunctionName() {
+    return 'custom-resource-apigw-account-cw-role';
```
Comment (alexdebrie) [inline lib/plugins/aws/lib/naming.js]: Should we make this shorter to give more room for service + stage name?

Do you know what our current effective limit is for service + stage combo, given other resources we create? We should try to not shorter than any further.

File: lib/plugins/aws/lib/naming.test.js [hunk 4]
```diff
@@ -468,18 +468,6 @@ describe('#naming()', () => {
     });
   });
 
-  describe('#getApiGatewayLogsRoleLogicalId()', () => {
```
Comment (alexdebrie) [inline lib/plugins/aws/lib/naming.test.js]: Do we want tests for the new naming functions?

File: lib/plugins/aws/customResources/index.js (hunk 1 above)
Comment (medikoo) [reply lib/plugins/aws/customResources/index.js]: > Would this happen if the service + stage name combo is too long?

I believe so.

This change is purely about exposing that error earlier than it happens now (during CF deployment).

I plan to refactor a custom resources handling, and with that I'll ensure we're more reliable here, but I don't want to focus on that in context of that PR

File: lib/plugins/aws/customResources/resources/apiGatewayCloudWatchRole/handler.js (hunk 2 above)
Comment (medikoo) [reply lib/plugins/aws/customResources/resources/apiGatewayCloudWatchRole/handler.js]: Thanks, I've taken that in.

Interestingly this string is not really used. I'd also clean this up, with upcoming refactor

File: lib/plugins/aws/lib/naming.js (hunk 3 above)
Comment (medikoo) [reply lib/plugins/aws/lib/naming.js]: I stripped the `account` to make it safer

File: lib/plugins/aws/lib/naming.test.js (hunk 4 above)
Comment (medikoo) [reply lib/plugins/aws/lib/naming.test.js]: Added

**Pull request merged**